qemu/tests/vioinput_hotplug.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import time
+
+from virttest import error_context
+from provider import input_tests
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test hotplug/unplug of virtio input device
+    1) Boot up w/ one virtio input device
+    2) Unplug one virtio input device
+    3) Hotplug one virtio input device
+    4) Run basic keyboard/mouse test
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    def hotplug_input_dev(vm, dev):
+        error_context.context("Hotplug %s" % dev, logging.info)
+        out, ver_out = vm.devices.simple_hotplug(dev, vm.monitor)
+        time.sleep(5)
+        if not ver_out:
+            test.fail("No % device in qtree after hotplug" % dev)
+        logging.info("%s is hotpluged successfully" % dev)
+
+    def unplug_input_dev(vm, dev):
+        error_context.context("Unplug %s" % dev, logging.info)
+        out, ver_out = vm.devices.simple_unplug(dev, vm.monitor)
+        time.sleep(15)
```

Comment:
And here.

Reply:
I'll move this line, it's my fault, I want to do plug after unplug before, but then found no need to do it, forget delete this line.